### PR TITLE
feat(workout mirroring): Add support to `HKHealthStore`'s `startWatchApp` method

### PR DESCRIPTION
**Pull request by gentilijuanmanuel**
### This is a...

✨ **`feat`** - New behavior has been implemented

### Context

- [Epic](https://runna.atlassian.net/browse/APP-7617)
- [Task](https://runna.atlassian.net/browse/APP-7619)
- Links to docs/ADRs (e.g., Notion, etc):
  - [Workout Mirroring - ADR](https://www.notion.so/runna/Mirror-workout-121bbd1a448580f4abadda99911e0b27?pvs=4)

### Description

This is the first PR to enable workout mirroring between our watchOS and mobile apps. What I'm trying to accomplish here is to enable the `HKHealthStore`'s `startWatchApp` method inside the React Native library we use to make the communication between the mobile app and `HealthKit` possible. Reason? We need to call this method when the user starts the workout from mobile app and we need to start the watchOS app session in parallel.

_Note_: we will need the same PR for [this one](https://developer.apple.com/documentation/healthkit/hkhealthstore/4172878-workoutsessionmirroringstarthand), which is the method used to get workout updates from watchOS app and reflect them in the mobile app UI.

**⚙️ Changes - what has been changed and why?**:

- Added `startWatchApp` as a new method to expose from the bridging header.
- Added `startWatchApp` TS function, following the same API pattern for library usage.
- Added `HealthKit` types that are used as parameters, such as `HKWorkoutConfiguration`.

### CODE REVIEW

File: src/native-types.ts [hunk 1]
```diff
@@ -2008,6 +2008,23 @@ type QueryWorkoutSamplesWithAnchorResponseRaw<
   readonly newAnchor: string
 }
 
+/**
+ * @see {@link https://developer.apple.com/documentation/healthkit/hkworkoutconfiguration Apple Docs }
+ */
+export type HKWorkoutConfiguration = {
+  readonly activityType: HKWorkoutActivityType;
+  readonly locationType?: HKWorkoutSessionLocationType;
```
Comment (walterholohan) [inline src/native-types.ts:2016]: Should this be optional?

File: src/utils/startWatchApp.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Native from '../native-types'
+
+import type { HKWorkoutConfiguration } from '..'
+
+const startWatchApp = (configuration: HKWorkoutConfiguration) => async () => Native.startWatchAppWithWorkoutConfiguration(configuration)
```
Comment (walterholohan) [inline src/utils/startWatchApp.ts:5]: Do we need to case the return value of `Native.startWatchAppWithWorkoutConfiguration(configuration)` to boolean 

e.g.
```
const startWatchApp = (configuration: HKWorkoutConfiguration) => async () => Native.startWatchAppWithWorkoutConfiguration(configuration) as Promise<boolean>
```

File: ios/ReactNativeHealthkit.swift [hunk 3]
```diff
@@ -2003,4 +2003,27 @@ class ReactNativeHealthkit: RCTEventEmitter {
...
+
+    let configuration = parseWorkoutConfiguration(workoutConfiguration)
+
+    store.startWatchApp(with: configuration) { success, error in
+      if let error {
+        reject(INIT_ERROR, INIT_ERROR_MESSAGE, error)
+        return
+      }
+
+      resolve(success)
```
Comment (walterholohan) [inline ios/ReactNativeHealthkit.swift:2025]: Is success a boolean?

**Review by walterholohan — APPROVED** [review]

File: ios/ReactNativeHealthkit.swift (hunk 3 above)
Comment (gentilijuanmanuel) [reply ios/ReactNativeHealthkit.swift:2025]: Yep!

Screenshot 2024-10-28 at 6 16 30 AM

Screenshot 2024-10-28 at 6 17 20 AM

File: src/native-types.ts (hunk 1 above)
Comment (gentilijuanmanuel) [reply src/native-types.ts:2016]: It's our decision, if we do not specify a location type, the `unknown` case will be taken:

```swift
/**
 @enum          HKWorkoutSessionLocationType
 @abstract      This enumerated type is used to represent the location type of a workout session.
 @discussion    This value represents whether a workout is performed indoors or outdoors.
 */
@available(iOS 10.0, *)
public enum HKWorkoutSessionLocationType : Int, @unchecked Sendable {

    case unknown = 1

    case indoor = 2

    case outdoor = 3
}
```

I can make it required!

File: src/utils/startWatchApp.ts (hunk 2 above)
Comment (gentilijuanmanuel) [reply src/utils/startWatchApp.ts:5]: `startWatchAppWithWorkoutConfiguration` (declared in `native-types.ts`) is already returning a `Promise<boolean>`:

```javascript
   /**
   * @see {@link https://developer.apple.com/documentation/healthkit/hkhealthstore/1648358-startwatchapp Apple Docs }
   */
  readonly startWatchAppWithWorkoutConfiguration: (
    workoutConfiguration: HKWorkoutConfiguration
  ) => Promise<boolean>;
  ```

File: ios/Constants.swift [hunk 4]
```diff
@@ -23,6 +23,9 @@ let HKWorkoutTypeIdentifier = "HKWorkoutTypeIdentifier"
 let HKWorkoutRouteTypeIdentifier = "HKWorkoutRouteTypeIdentifier"
 let HKDataTypeIdentifierHeartbeatSeries = "HKDataTypeIdentifierHeartbeatSeries"
 
+let HKWorkoutActivityType_PROPERTY_NAME = "activityType"
+let HKWorkoutSessionLocationType_PROPERTY_NAME = "locationType"
```
Comment (fnazarios) [inline ios/Constants.swift:27]: nit: Maybe we should use the same indentation here. 
```suggestion
let HKWorkoutActivityTypePropertyName = "activityType"
let HKWorkoutSessionLocationTypePropertyName = "locationType"
```

**Review by fnazarios — APPROVED** [review]
LGTM

File: ios/Constants.swift (hunk 4 above)
Comment (gentilijuanmanuel) [reply ios/Constants.swift]: Fixed, thanks!

**Pull request merged**